contracts/chain-adapters/Arbitrum_LegacyCustomGasToken_Adapter.sol

```diff
@@ -0,0 +1,339 @@
+                // amount and requiredL1TokenTotalFeeAmount are in the precision of the custom gas token.
+                uint256 amountToBridge = amount + requiredL1TokenTotalFeeAmount;
+                CUSTOM_GAS_TOKEN.safeIncreaseAllowance(address(L1_INBOX), amountToBridge);
+                L1_INBOX.createRetryableTicket(
+                    to, // destAddr destination L2 contract address
+                    amount, // l2CallValue call value for retryable L2 message
+                    L2_MAX_SUBMISSION_COST, // maxSubmissionCost Max gas deducted from user's L2 balance to cover base fee
+                    L2_REFUND_L2_ADDRESS, // excessFeeRefundAddress maxgas * gasprice - execution cost gets credited here on L2
+                    L2_REFUND_L2_ADDRESS, // callValueRefundAddress l2Callvalue gets credited here on L2 if retryable txn times out or gets cancelled
+                    RELAY_MESSAGE_L2_GAS_LIMIT, // maxGas Max gas deducted from user's L2 balance to cover L2 execution
+                    L2_GAS_PRICE, // gasPriceBid price bid for L2 execution
+                    amountToBridge, // tokenTotalFeeAmount amount of fees to be deposited in native token.
+                    "0x" // data ABI encoded data of L2 message
+                );
+            } else {
+                IERC20(l1Token).safeIncreaseAllowance(erc20Gateway, amount);
+                CUSTOM_GAS_TOKEN.safeIncreaseAllowance(erc20Gateway, requiredL1TokenTotalFeeAmount);
+
+                // To pay for gateway outbound transfer with custom gas token, encode the tokenTotalFeeAmount in the data field:
+                // The data format should be (uint256 maxSubmissionCost, bytes extraData, uint256 tokenTotalFeeAmount).
+                // Source: https://github.com/OffchainLabs/token-bridge-contracts/blob/5bdf33259d2d9ae52ddc69bc5a9cbc558c4c40c7/contracts/tokenbridge/ethereum/gateway/L1OrbitERC20Gateway.sol#L57
+                bytes memory data = abi.encode(L2_MAX_SUBMISSION_COST, "", requiredL1TokenTotalFeeAmount);
+                L1_ERC20_GATEWAY_ROUTER.outboundTransferCustomRefund(
+                    l1Token,
+                    L2_REFUND_L2_ADDRESS,
+                    to,
+                    amount,
+                    RELAY_TOKENS_L2_GAS_LIMIT,
+                    L2_GAS_PRICE,
+                    data
+                );
+            }
+        }
+        emit TokensRelayed(l1Token, l2Token, amount, to);
+    }
+
+    /**
+     * @notice Returns required amount of gas token to send a message via the Inbox.
+     * @dev Should return a value in the same precision as the gas token's precision.
+     * @param l2GasLimit L2 gas limit for the message.
+     * @return amount of gas token that this contract needs to hold in order for relayMessage to succeed.
+     */
+    function getL1CallValue(uint32 l2GasLimit) public view returns (uint256) {
+        return L2_MAX_SUBMISSION_COST + L2_GAS_PRICE * l2GasLimit;
```

Comment:
Changed this to not use `_from18ToNativeTokenDecimals` under the assumption that when the deployer deploys this adapter, they will correctly set `L2_MAX_SUBMISSION_COST` and `L2_GAS_PRICE` to the correct precision.